### PR TITLE
Add whitespace analyzers

### PR DESCRIPTION
**Pull request by pakrym**
Adds two whitespace related rules:

1. Avoid multiple empty lines in a row
2. Avoid whitespace in the end of the line

### CODE REVIEW

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs [hunk 1]
```diff
@@ -40,5 +40,12 @@ internal class Descriptors
         public static DiagnosticDescriptor AZC0008 = new DiagnosticDescriptor(
             "AZC0008", "ClientOptions should have a nested enum called ServiceVersion",
             "Client type should have a nested enum called ServiceVersion", "Usage", DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor AZC0101 = new DiagnosticDescriptor(
```
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs]: I started non-guideline-specific rules with an offset of 100.

Comment (annelo-msft) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs]: That was going to be my first question! :)

Is there any risk that the number of guideline-specific rules will exceed 100?

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(new[]
+        {
+            Descriptors.AZC0101,
+            Descriptors.AZC0102
+        });
+    }
+}
```
Comment (annelo-msft) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs]: Any concerns about there not being a newline at the EOF?

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.ClientSdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NewLineAnalyzer: DiagnosticAnalyzer
```
Comment (annelo-msft) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs]: NewLineAnalyzer is checking for  trailing whitespace as well as new lines ... is there a name that would encompass both?

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs (hunk 1 above)
Comment (pakrym) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs]: I'll make it 1000.

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs (hunk 2 above)
Comment (pakrym) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/NewLineAnalyzer.cs]: We are not really enforcing that but maybe we should.

**Comment by annelo-msft** [comment]
Would it make sense to rename the files to match the new class names?  (For the tests and analyzer)

**Pull request merged**